src/cosalib/gcp.py

```diff
@@ -39,36 +39,56 @@ def gcp_run_ore(build, args):
     if args.project is None:
         raise Exception(arg_exp_str.format("project", "GCP_PROJECT"))
 
-    ore_args = ['ore']
-    if args.log_level == "DEBUG":
-        ore_args.extend(['--log-level', "DEBUG"])
-
     gcp_name = re.sub(r'[_\.]', '-', build.image_name_base)
     if not re.fullmatch(GCP_NAMING_RE, gcp_name):
         raise Exception(f"{gcp_name} does match the naming rule: file a bug")
-
     urltmp = os.path.join(build.tmpdir, "gcp-url")
-    ore_args.extend([
+
+    ore_common_args = [
+        'ore',
         'gcloud',
         '--project', args.project,
-        '--basename', build.build_name,
+        '--json-key', args.json_key,
+
+    ]
+    if args.log_level == "DEBUG":
+        ore_common_args.extend(['--log-level', "DEBUG"])
+
+    ore_upload_cmd = ore_common_args + [
         'upload',
+        '--basename', build.build_name,
         '--force',  # We want to support restarting the pipeline
         '--bucket', f'{args.bucket}',
-        '--json-key', args.json_key,
         '--name', gcp_name,
         '--file', f"{build.image_path}",
         '--write-url', urltmp,
-    ])
-
-    if args.family:
-        ore_args.extend(['--family', args.family])
+    ]
     if args.description:
-        ore_args.extend(['--description', args.description])
+        ore_upload_cmd.extend(['--description', args.description])
     if not args.create_image:
-        ore_args.extend(['--create-image=false'])
+        ore_upload_cmd.extend(['--create-image=false'])
+    run_verbose(ore_upload_cmd)
+
+    # Run deprecate image to deprecate if requested
+    if args.deprecated:
+        ore_deprecate_cmd = ore_common_args + [
+            'deprecate-image',
+            '--image', gcp_name,
+            '--state', 'DEPRECATED'
+        ]
+        run_verbose(ore_deprecate_cmd)
+
+    # Run update-image to add to an image family if requested.
+    # We run this as a separate API call because we want to run
+    # it AFTER the deprecation if the user passed --deprecated
+    if args.family:
+        ore_update_cmd = ore_common_args + [
+            'update-image',
```

Comment:
In principle the deprecation could also happen as part of `update-image`, saving an API call, right?

Not a big deal though.

Reply:
Right. Let's discuss in https://github.com/coreos/coreos-assembler/pull/1444#discussion_r423954918

Reply:
https://github.com/coreos/coreos-assembler/pull/1444#issuecomment-628921354